message_app/db/db.py

```diff
@@ -0,0 +1,56 @@
+import sqlite3
+import os
+from flask import current_app, g
+
+# Connect to the database
+# The g name stands for “global”, but that is referring to the data being global within a context.
+# The data on g is lost after the context ends. Flask provides the g object for this purpose.
+# g is a simple namespace object that has the same lifetime as an application context.
+# db is an attribute of object g. We will store the connection to our database to g.db
+# The sqlite3.Row is used to convert the row data in tuple to dictionary for easier access
+def get_db():
+    if 'db' not in g:
+        g.db = sqlite3.connect(
+            current_app.config['DATABASE'],
+            detect_types=sqlite3.PARSE_DECLTYPES
+        )
+        g.db.row_factory = sqlite3.Row
+
+    return g.db
+
+
+# Close the database connection
+# We close the connection to our database by removing it from the g object
+def close_db(e=None):
+    db = g.pop('db', None)
+
+    if db is not None:
+        db.close()
+
+
+# Run a query
+# When we pass variable parts to the SQL statement, we will avoid directly adding them
+# to the SQL statement with string formatting because this makes it possible to attack
+# the application using SQL Injections. Instead, we will pass in arguments as a dictionary
+# The parameter one is used for fetching the first row.
+def query_db(query: str, args={}, one=False):
+    cur = get_db().execute(query, args)
+    rv = cur.fetchall()
+    cur.close()
+    return (rv[0] if rv else None) if one else rv
```

Comment:
This whole part is subject to future changes (it might be deleted) when we introduce `sqlalchemy `and data models to our app, right?

Reply:
Yes. This part is used only to avoid SQL injection. When we use data models, it will validate the input for us, and this function is no longer necessary.